### DISCUSSION TITLE
Unexpected client ip behavior

### DISCUSSION THREAD

**Discussion by ghost** [uNetworking/uWebSockets.js#402] [Q&A] [answered]
I'm running an uws server on localhost. Whenever I retrieve the clients ip address, it throws 127.0.0.1. I'm expecting it to throw my public ipv4. Why's that?

Note: I'm using the same device for both client and server.

**Comment by hst-m** [comment, marked as answer]
127.0.0.1 is special address for localhost, this is correct, it means it is coming from the same machine. it would only show the external ip if it was coming from outside. this is basic networking 101

https://www.lifewire.com/network-computer-special-ip-address-818385
